test/extensions/filters/http/ext_proc/mutation_utils_test.cc

```diff
@@ -244,6 +245,55 @@ TEST(MutationUtils, TestBodyMutationNothing) {
   EXPECT_TRUE(TestUtility::buffersEqual(buf, bufCopy));
 }
 
+TEST(MutationUtils, TestAllowHeadersExactCaseSensitive) {
```

Comment:
Just tested EXACT case to verify the headersToProto() utility function is honoring the new header_matchers parameter, and working as expected.  The complete matching function test is in this file:
https://github.com/envoyproxy/envoy/blob/c2ae2211196a48b12d2e36d00c6c2889ae2f434a/test/common/common/matchers_test.cc#L315